### PR TITLE
Fix anti-pattern in SelectorInput

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
https://github.com/red-hat-storage/odf-console/issues/671

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@SanjalKatiyar Please review

**Comment by SanjalKatiyar** [comment]
/hold
@TimothyAsirJeyasing as discussed earlier, please add a small screencast of the working of this component in "apply-policy" modal of MCO. (This FC is used multiple times on the same modal, add PVC labels to all the input boxes on the modal and check/verify functionality is working as expected).

File: packages/shared/modals/Selector.tsx [hunk 1]
```diff
@@ -170,11 +170,11 @@ export class SelectorInput extends React.Component<
...
 
-  static getDerivedStateFromProps(props, state) {
-    if (!_.isEqual(props.tags, state.tags)) {
-      return { ...state, tags: props.tags };
+  componentDidUpdate(prevProps) {
+    if (!_.isEqual(prevProps.tags, this.props.tags)) {
+      return { ...this.state, tags: this.props.tags };
     }
-    return state;
+    return this.state;
```
Comment (SanjalKatiyar) [inline packages/shared/modals/Selector.tsx:177]: nits:
```suggestion
```

Comment (SanjalKatiyar) [inline packages/shared/modals/Selector.tsx:177]: `ComponentDidUpdate` should not return anything...

Comment (SanjalKatiyar) [reply packages/shared/modals/Selector.tsx:177]: check the React docs...

File: packages/shared/modals/Selector.tsx [hunk 2]
```diff
@@ -170,11 +170,14 @@ export class SelectorInput extends React.Component<
...
-    if (!_.isEqual(props.tags, state.tags)) {
-      return { ...state, tags: props.tags };
+  componentDidUpdate(prevProps) {
+    if (!_.isEqual(prevProps.tags, this.props.tags)) {
+      this.updateProp(this.props.tags);
     }
-    return state;
+  }
+
+  updateProp(tags: string[]) {
```
Comment (alfonsomthd) [inline packages/shared/modals/Selector.tsx:179]: Shouldn't this be called `updateTags` ?
```suggestion
  priavte updateTags(tags: string[]) {
```

File: packages/shared/modals/Selector.tsx [hunk 3]
```diff
@@ -170,11 +170,14 @@ export class SelectorInput extends React.Component<
...
-      return { ...state, tags: props.tags };
+  componentDidUpdate(prevProps) {
+    if (!_.isEqual(prevProps.tags, this.props.tags)) {
+      this.updateTags(this.props.tags);
     }
-    return state;
+  }
+
+  private updateTags(tags: string[]) {
+    this.setState({ tags: tags });
```
Comment (SanjalKatiyar) [inline packages/shared/modals/Selector.tsx:180]: make it inline, no need for separate function... and test for mco apply-policy modal once again, similar to how we did earlier...

**Comment by TimothyAsirJeyasing** [comment]
https://user-images.githubusercontent.com/6670284/229709073-c7766234-7d3e-488c-89e8-50d385b04f55.mp4

File: packages/shared/modals/Selector.tsx (hunk 3 above)
Comment (TimothyAsirJeyasing) [reply packages/shared/modals/Selector.tsx:180]: i received an lint error when i keep inside saying " Do not use setState in componentDidUpdate". I have seen in console they kept outside function.

Comment (SanjalKatiyar) [reply packages/shared/modals/Selector.tsx:180]: "i received an lint error when i keep inside saying " Do not use setState in componentDidUpdate" >> this does not solve anything, u r still using "setState" in "componentDidUpdate", just now calling it from another location...
"I have seen in console they kept outside function." >> can u send me console link to this ??

Comment (alfonsomthd) [reply packages/shared/modals/Selector.tsx:180]: @SanjalKatiyar I think Tim refers to [this](https://github.com/openshift/console/blob/master/frontend/public/components/utils/selector-input.jsx#L27).

Comment (SanjalKatiyar) [reply packages/shared/modals/Selector.tsx:180]: yes, I checked that...
I meant here that there is no need to create a separate function just to add single line "this.setState({ tags: tags });",
functions are created for either re-usability purpose or to make  "componentDidUpdate" clean... but here "updateTags" is only a single line, single use function which can be made inline in "componentDidUpdate"...

anyway... not a hard requirement, consider this comment as "nits", up to the assignee to fix if what, else no need as currently it is alright as well...

**Comment by TimothyAsirJeyasing** [comment]
https://user-images.githubusercontent.com/6670284/229754353-8d164b9a-db45-407a-9d58-2a77d04e9dd1.mp4

**Pull request merged**